### PR TITLE
Adds JANITOR Processor

### PR DESCRIPTION
**Pull request by Miserlou**
## Issue Number

Despite our best efforts, our EBS volumes were filling up with trash.

## Purpose/Implementation Notes

Adds a new processor, JANITOR, which will clean up jobs which get killed by OOM or AWS interruption.

## Methods

Dispatched by the foreman, runs every 30 minutes on each volume.

## Types of changes
- New feature (non-breaking change which adds functionality)

## Functional tests
New tests added.

### CODE REVIEW

File: workers/data_refinery_workers/processors/utils.py [hunk 1]
```diff
@@ -76,50 +76,56 @@ def start_job(job_context: Dict):
...
-        job_context["original_files"] = original_files
-        original_file = job_context['original_files'][0]
-        assocs = OriginalFileSampleAssociation.objects.filter(original_file=original_file)
-        samples = Sample.objects.filter(id__in=assocs.values('sample_id')).distinct()
-        job_context['samples'] = samples
-        job_context["computed_files"] = []
+    # Janitors have no requirement
+    if job.pipeline_applied not in ["JANITOR"]:
+        # Some jobs take OriginalFiles, other take Datasets
+        if job.pipeline_applied not in ["SMASHER", "QN_REFERENCE"]:
```
[the remark this answers is not in the document]
Comment (Miserlou) [reply workers/data_refinery_workers/processors/utils.py]: Refactored

File: foreman/data_refinery_foreman/foreman/test_main.py [hunk 2]
```diff
@@ -512,3 +512,20 @@ def test_retrying_lost_processor_jobs_time(self, mock_send_job):
...
+
+        main.send_janitor_jobs(range(0,60))
+
+        self.assertEqual(ProcessorJob.objects.all().count(), 6)
+        self.assertEqual(ProcessorJob.objects.filter(pipeline_applied="JANITOR").count(), 3)
+
+        ixs = ["1", "2", "3"]
+        for p in ProcessorJob.objects.filter(pipeline_applied="JANITOR"):
+            self.assertTrue(p.volume_index in ixs)
+            ixs.remove(p.volume_index)
```
[the remark this answers is not in the document]
Comment (Miserlou) [reply foreman/data_refinery_foreman/foreman/test_main.py]: Added comment

File: workers/nomad-job-specs/janitor.nomad.tpl [hunk 3]
```diff
@@ -0,0 +1,95 @@
+job "JANITOR_${{INDEX}}_${{RAM}}" {
```
[the remark this answers is not in the document]
Comment (Miserlou) [reply workers/nomad-job-specs/janitor.nomad.tpl]: We don't, but then we'd also need to refactor how we create processor jobs and it doesn't hurt to have the flexibility.

[the remark this answers is not in the document]
Comment (kurtwheeler) [reply workers/nomad-job-specs/janitor.nomad.tpl]: We already don't do this for smasher and downloader jobs, but it doesn't really hurt anything to have it I guess.

**Pull request merged**